homeassistant/components/loqed/sensor.py

```diff
@@ -12,16 +14,36 @@
     PERCENTAGE,
     SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
     EntityCategory,
+    UnitOfElectricPotential,
 )
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
 
-from .const import DOMAIN
+from .const import BATTERY_TYPES, DOMAIN, EVENT_TYPES
 from .coordinator import LoqedDataCoordinator, StatusMessage
 from .entity import LoqedEntity
 
-SENSORS: Final[tuple[SensorEntityDescription, ...]] = (
-    SensorEntityDescription(
+
+@dataclass
+class LoqedSensorEntityDescription(SensorEntityDescription):
+    """Dataclass describing LOQED sensor entities."""
+
+    transform_value: Callable[[StateType], StateType] = lambda data: data
+
+
+SENSORS: Final[tuple[LoqedSensorEntityDescription, ...]] = (
+    LoqedSensorEntityDescription(
+        key="last_event",
+        translation_key="last_event",
+        options=EVENT_TYPES,
+        transform_value=lambda data: data if data != "" else "unknown",
```

Comment:
The result of an unknown item in an enum is already "unknown" automatically, this logic is thus not needed.

```suggestion
```

Reply:
Unfortunately default value is `""` (empty string). It causes an exception.

Reply:
In case, we need to map an empty string to `None` (as in a `NoneType`), not `unknown`.

Reply:
Makes sense. I'll update PR.

Reply:
Should be fixed.